File: SynapseBackend/SynapseBackend/Entities/Flashcard.cs
using System.ComponentModel.DataAnnotations;

namespace SynapseBackend.Entities;

public class Flashcard
{
    [Key]
    public Guid Id { get; set; }
    public Guid DeckId { get; set; }
    
    [MaxLength(512)]
    public required String Front { get; set; }
    [MaxLength(1024)]
    public required String Back { get; set; }
    public long LastModified { get; set; } = DateTimeOffset.UtcNow.ToUnixTimeMilliseconds();
    public bool IsDeleted { get; set; } = false;

    public Deck Deck { get; set; } = null!;
}

File: synapse_mobile/app/src/main/java/com/olidev/synapse_mobile/data/local/entities/Flashcard.kt
package com.olidev.synapse_mobile.data.local.entities

import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.Index
import androidx.room.PrimaryKey
import java.util.UUID

@Entity(
    tableName = "flashcards",
    foreignKeys = [
        ForeignKey(
            entity = Deck::class,
            parentColumns = ["id"],
            childColumns = ["deckId"],
            onDelete = ForeignKey.CASCADE
        )
    ],
    indices = [Index("deckId")]
)
data class Flashcard(
    @PrimaryKey val id: String = UUID.randomUUID().toString(),
    val deckId : String,
    val front: String,
    val back: String,
    val isSynced : Boolean = false,
    val lastModified : Long = System.currentTimeMillis(),
    val isDeleted : Boolean = false
)
